lib/push.js

```diff
@@ -35,25 +44,31 @@ module.exports = function (log, db) {
             if (device.pushCallback) {
               // send the push notification
               log.increment(LOG_UPDATE_INCREMENT_SEND)
-              request.post(device.pushCallback, function (err){
+              request.post(device.pushCallback, function (err, response) {
                 if (err) {
-                  log.error({
-                    op: LOG_OP_NOTIFY_UPDATE,
-                    deviceId: deviceId,
-                    err: err
-                  })
-                  log.increment(LOG_UPDATE_INCREMENT_FAIL)
+                  // any 400 level error from the push servers means
+                  // the push settings need to be reset.
+                  // the clients will check this and re-register push endpoints
```

Comment:
I think it makes sense to only reset if `statusCode == 404 || statusCode == 410`. The Mozilla server currently returns a 404 for nonexistent and expired subscriptions; mozilla-services/autopush#312 proposes 410 instead. You'll only get a 413 if you send an encrypted payload exceeding 4 KB. We don't currently use 429.